### DISCUSSION TITLE
Lable predictions directly with .ONNX model

### DISCUSSION THREAD

**Discussion by pasmai** [ultralytics/yolov5#5692] [Q&A]
Hello everyone

I would like that when predicting a `.onnx` model that the labels are also directly shown in the image.

`python detect.py --weights /models/trained/yolov5m.pt` generates an image that contains labels.

`python detect.py --weights /models/trained/yolov5m.onnx` generates an image that where the boxes are labelled as `class0`
![zidane](https://user-images.githubusercontent.com/14838120/142244137-4a51a407-a3f9-49fe-84fc-f79f3a495673.jpg)

Is there a good way to re-use or include the labels from the `.yaml` datasets?

**Comment by glenn-jocher** [comment]
@pasmai ONNX models don't have class names attached as metadata, but you can insert your own class names in DetectMultiBackend():
https://github.com/ultralytics/yolov5/blob/562191f5756273aca54225903f5933f7683daade/models/common.py#L294
